server/base/src/main/java/org/apache/accumulo/server/security/AuditedSecurityOperation.java

```diff
@@ -622,10 +622,10 @@ public void grantSystemPermission(TCredentials credentials, String user,
 
   @Override
   public void grantTablePermission(TCredentials credentials, String user, TableId tableId,
-      TablePermission permission, NamespaceId namespaceId) throws ThriftSecurityException {
-    String tableName = getTableName(tableId);
```

Comment:
Is this what was causing the performance problem?

Reply:
Yes, the code down this path is an issue.